src/client/datascience/types.ts

```diff
@@ -590,6 +591,7 @@ export interface INotebookEditor extends Disposable, IInteractiveBase {
      * Once VSC Notebook is stable, this property can be removed.
      */
     readonly type: 'old' | 'custom' | 'native';
+    readonly document: NotebookDocument | undefined;
```

Comment:
I'd remove this. 
Basically what this does is we're prolonging the life of `INotebookEditor`.
I'd prefer we don't use such wrappers where possible.
E.g. WebView & Interactive windows don't need this property (yes I can see that from `undefined`.
But we're just adding more stuff to some wrappers and it just keeps getting used elsewheree.